### DISCUSSION TITLE
vscode java support byte array memory view or equivalent "memory dump" watch view, see bytes/chars.

### DISCUSSION THREAD

**Discussion by AshleyT3** [microsoft/vscode-java-debug#1310] [Ideas]
Environment: Based on my using VSCode 1.76 on Linux w/Java (17) and VSCode Java debugging...

Request for VSCode Java debugging to support typical debugger capability of having a watch window, or memory dump-style view that shows a byte array's raw bytes hex values with characters, or at least just character representation similar to what most debuggers have. For example, WinDbg supports "db" to dump bytes, Visual Studio can do the same, even for managed.

This is not critical for me but just wanted to create a placeholder request for others to +1 in case it ever seems a worthy feature (and in case I missed that this already exists... forgive in advance if so!).

Workaround...

I found an ancient non-vscode [Stack Overflow article/answer](https://stackoverflow.com/questions/381546/how-can-i-display-a-byte-array-as-a-string-in-javas-debug-detail-formatter) with an answer viable for use in the VSCode debugger, just wrap the array in String object to get the ASCII view...

```
// if code has this...
byte[] bytes = {0x41, 0x42, 0x43, 0x31, 0x32, 0x33};
// can do this in debugger console...
new String(bytes)
"ABC123"
```

Cumbersome but fine for my immediate needs. I do not currently work daily in Java.

**Comment by montematheus50-boop** [comment]
expancao ta errada
